ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java

```diff
@@ -59,14 +59,8 @@
 import com.arjuna.ats.internal.arjuna.common.UidHelper;
 import com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord;
 import com.arjuna.ats.jta.common.jtaPropertyManager;
-import com.arjuna.ats.jta.logging.RecoveryRequired;
 import com.arjuna.ats.jta.logging.jtaLogger;
-import com.arjuna.ats.jta.recovery.SerializableXAResourceDeserializer;
-import com.arjuna.ats.jta.recovery.XARecoveryResource;
-import com.arjuna.ats.jta.recovery.XARecoveryResourceManager;
-import com.arjuna.ats.jta.recovery.XAResourceOrphanFilter;
-import com.arjuna.ats.jta.recovery.XAResourceRecovery;
-import com.arjuna.ats.jta.recovery.XAResourceRecoveryHelper;
+import com.arjuna.ats.jta.recovery.*;
```

Comment:
We have a checkstyle plugin which is skipped by default because we have a lot of old files, that we don't want to change, which would fail the compilation. But on new code we do encourage/advise that it should follow our checkstyle rules.

To see if your code follows our style guide you have two options:

1. Enable the checkstyle plugin via the [pom file](https://github.com/jbosstm/narayana/blob/master/pom.xml#L377), or

2. Reference the rule file from your IDE settings (for Intelij that would be Settings -> Other Settings -> Checkstyle) and we use the [wildfly rule set](https://github.com/wildfly/wildfly-checkstyle-config/tree/master/src/main/resources/wildfly-checkstyle).

Reply:
> We can avoid the new RecoveryRequired class and just add an AtomicBoolean to this class instead.

Okay, I will remove the new RecoveryRequired class and add AtomicBoolean

Reply:
> We have a checkstyle plugin which is skipped by default because we have a lot of old files, that we don't want to change, which would fail the compilation. But on new code we do encourage/advise that it should follow our checkstyle rules.
> 
> To see if your code follows our style guide you have two options:
> 
> 1. Enable the checkstyle plugin via the [pom file](https://github.com/jbosstm/narayana/blob/master/pom.xml#L377), or
> 2. Reference the rule file from your IDE settings (for Intelij that would be Settings -> Other Settings -> Checkstyle) and we use the [wildfly rule set](https://github.com/wildfly/wildfly-checkstyle-config/tree/master/src/main/resources/wildfly-checkstyle).

Maybe checkstyle changes can be done on separate PR?

Reply:
The checkstyle issue I commented on was introduced as part of this PR so a separate PR would not make sense.

I admit that having to manually run the checkstyle plugin is not optimal. But I do not see an alternative to the reason I gave:

>  We have a checkstyle plugin which is skipped by default because we have a lot of old files, that we don't want to change, which would fail the compilation. But on new code we do encourage/advise that it should follow our checkstyle rules.

@ochaloup We have had this problem of  skipping the checkstyle plug on two PRs this week, it mainly affects occasional contributors. Do you have thoughts on how we could improve it (other than fixing all of the historical code).

Reply:
I see that your latest push has fixed the checkstyle problem I reported.
But I will not mark the conversation as resolved yet since I have an outstanding question for Ondra.

Reply:
Sorry for not reacting this sooner I wanted but I haven't got to this. I can see the discussion was moved to chat. I replied there.